server/lib/file-upload/aws.ts

```diff
@@ -101,33 +111,34 @@ export default class Aws implements FileStore {
       Bucket: this.bucket,
     }
 
-    const result = await this.client.getObject(params).promise()
-    // NOTE(tec27): AWS docs say this will be a Buffer in Node, types are weird though
-    return (result.Body as Buffer) ?? Buffer.alloc(0)
+    const result = await this.client.getObject(params)
+    if (!result.Body) {
+      return Buffer.alloc(0)
+    }
+    return Buffer.from(await result.Body.transformToByteArray())
   }
 
   // Options object can contain any of the valid keys specified in the AWS SDK for the
   // `deleteObject` function.
   async delete(filename: string, options: any = {}) {
     const normalized = this.getNormalizedPath(filename)
     const params = { Key: normalized, Bucket: this.bucket, ...formatOptions(options) }
-    return this.client.deleteObject(params).promise()
+    return this.client.deleteObject(params)
   }
 
   // Options object can contain any of the valid keys specified in the AWS SDK for the
   // `deleteObjects` function.
   async deleteFiles(prefix: string, options: any = {}) {
     const normalized = this.getNormalizedPath(prefix)
-    const files = await this.client
-      .listObjectsV2({ Prefix: normalized, Bucket: this.bucket })
-      .promise()
+    const files = await this.client.listObjectsV2({ Prefix: normalized, Bucket: this.bucket })
     const keys = (files.Contents ?? []).map(file => ({ Key: file.Key }))
     const params = { Delete: { Objects: keys }, Bucket: this.bucket, ...formatOptions(options) }
-    return this.client.deleteObjects(params).promise()
+    return this.client.deleteObjects(params)
   }
 
   url(filename: string, options: any = {}): string {
-    const url = `https://${this.bucket}.${this.endpoint}/${filename}`
+    const hostname = new URL(this.endpoint).hostname
```

Comment:
Given that the endpoint never changes, it feels nicer to just calculate this when this object is constructed and not re-parse the URL every time we need a url